kroxylicious-kms-provider-fortanix-dsm/src/test/java/io/kroxylicious/kms/provider/fortanix/dsm/session/ApiKeySessionProviderTest.java

```diff
@@ -214,16 +218,17 @@ void previousSessionTerminatedOnServer() {
             var sessionStage = provider.getSession();
             assertThat(sessionStage)
                     .succeedsWithin(Duration.ofSeconds(1))
-                    .satisfies(s -> assertThat(s.authorizationHeader()).isEqualTo("Bearer firstToken"));
+                    .satisfies(s -> assertThat(s.authorizationHeader()).isEqualTo("Bearer " + initialToken));
 
-            var refreshed = createTestCredential("Bearer", expiresInSecs, "secondToken");
+            var refreshed = createTestCredential("Bearer", expiresInSecs, secondToken);
             server.stubFor(
                     post(urlEqualTo(SESSION_AUTH_ENDPOINT))
                             .willReturn(aResponse().withBody(toJson(refreshed))));
 
             await().atMost(Duration.ofSeconds(5))
                     .untilAsserted(() -> {
-                        server.verify(moreThanOrExactly(1), postRequestedFor(urlEqualTo(SESSION_TERMINATE_ENDPOINT)));
+                        server.verify(1, postRequestedFor(urlEqualTo(SESSION_TERMINATE_ENDPOINT))
+                                .withHeader(AUTHORIZATION_HEADER, equalTo("Bearer " + initialToken)));
```

Comment:
I think the test should really be asserting the the first session is logged out and that happens exactly once.

Reply:
I haven't read the API docs but from how I read this change we should expect a new token after every refresh so there should only ever be 1 call to terminate session with a given token? 

I still question if that is actually the assertion we want to make: Where is the harm in us calling `terminate token X` multiple times? As I understand things we want to ensure we call it at least once for a fire and forget operation I don't think its reasonable to assert we only fire it once.